cmd/eib/main.go

```diff
@@ -24,43 +25,170 @@ import (
+	flag.StringVar(&cliArguments.rootBuildDir, argBuildDir, "", "full path to the directory to store build artifacts")
+	flag.BoolVar(&cliArguments.validate, argValidate, false, "if specified, the image definition will be validated but not built")
 	flag.Parse()
 
-	imageDefinition, err := parseImageDefinition(configFile, configDir)
+	return cliArguments
+}
+
+// Configures the global logger.
+func setupLogging(buildDir string) {
+	logFilename := filepath.Join(buildDir, logFilename)
+
+	logConfig := zap.NewProductionConfig()
+	logConfig.Level = zap.NewAtomicLevelAt(zap.DebugLevel)
+	logConfig.Encoding = "console"
+	logConfig.EncoderConfig.EncodeLevel = zapcore.CapitalLevelEncoder
+	logConfig.EncoderConfig.EncodeTime = zapcore.ISO8601TimeEncoder
+	logConfig.OutputPaths = []string{logFilename}
+
+	logger := zap.Must(logConfig.Build())
+
+	// Set our configured logger to be accessed globally by zap.L()
+	zap.ReplaceGlobals(logger)
+}
+
+// Returns whether the image configuration directory was specified and can be read, displaying
+// the appropriate messages to the user. Returns 'true' if the directory exists and execution can proceed,
+// 'false' otherwise.
+func doesImageConfigDirExist(cliArguments CLIArguments) bool {
+	configDir := cliArguments.configDir
+
+	if configDir == "" {
+		audit.AuditInfof("The '%s' argument must be specified.", argConfigDir)
+		return false
+	}
+
+	_, err := os.Stat(configDir)
 	if err != nil {
-		return nil, fmt.Errorf("parsing image definition file %s: %w", configFile, err)
+		if errors.Is(err, fs.ErrNotExist) {
+			audit.AuditInfof("The specified image configuration directory '%s' could not be found.", configDir)
+			return false
+		}
+		audit.AuditInfof("Unable to check the filesystem for the image configuration directory '%s'. %s",
+			configDir, checkLogMessage)
+		zap.S().Error(err)
+		return false
 	}
 
-	err = validateImageConfigDir(configDir)
+	return true
+}
+
+// Attempts to parse the specified image definition file, displaying the appropriate messages to the user.
+// Returns a populated `image.Context` struct if successful, `nil` if the definition could not be parsed.
+func parseImageDefinition(cliArguments CLIArguments) *image.Definition {
+	definitionFilePath := filepath.Join(cliArguments.configDir, cliArguments.definitionFile)
+
+	_, err := os.Stat(definitionFilePath)
 	if err != nil {
-		return nil, fmt.Errorf("validating the config dir %s: %w", configDir, err)
+		if errors.Is(err, fs.ErrNotExist) {
+			audit.AuditInfof("The specified definition file '%s' could not be found.", definitionFilePath)
+		} else {
+			audit.AuditInfof("Unable to check the filesystem for the image definition file '%s'. %s",
+				definitionFilePath, checkLogMessage)
+			zap.S().Error(err)
+		}
+		return nil
 	}
 
-	buildDir, combustionDir, err := build.SetupBuildDirectory(rootBuildDir)
+	configData, err := os.ReadFile(definitionFilePath)
```

Comment:
This will already return `fs.ErrNotExist` if the file doesn't exist so we can skip the `os.Stat` call if you'd like.